icu4j/main/classes/core/src/com/ibm/icu/impl/units/UnitPreferences.java

```diff
@@ -56,14 +70,54 @@ private static String[] getAllUsages(String usage) {
         return result.toArray(new String[0]);
     }
 
-    public UnitPreference[] getPreferencesFor(String category, String usage, String region) {
+    public UnitPreference[] getPreferencesFor(String category, String usage, ULocale locale, UnitsData data) {
+        // TODO: remove this condition when all the categories are allowed.
+        if (category.equals("temperature")) {
+            String localeUnit = locale.getKeywordValue("mu");
+            String localeUnitCategory;
+            try {
+                localeUnitCategory = localeUnit == null ? null : data.getCategory(MeasureUnitImpl.forIdentifier(localeUnit));
+            } catch (Exception e) {
+                localeUnitCategory = null;
+            }
+
+            if (localeUnitCategory != null && category.equals(localeUnitCategory)) {
+                UnitPreference[] preferences = {new UnitPreference(localeUnit, null, null)};
+                return preferences;
+            }
+        }
+
+        String region = locale.getCountry();
```

Comment:
Suggestion (optional): Make this variable a `CharSequence` and use `localeRegion.subSequence` below so that we can avoid allocating extra strings.